feat(agent): Expand /var/run volume mount to include entire /var

lrishi opened the pull request:
## What this PR does / why we need it:
Agent requires access to container fs for internal state validation and feature support. On some platforms, these directories might reside outside of /var/run in /var/<vendor-specific> directories.

## Checklist

- [x] Title of the PR starts with type and scope, (e.g. feat(agent,node-analyzer,sysdig-deploy):)
- [x] Chart Version bumped for the respective charts
- [x] Check GithubAction checks (like lint) to avoid merge-check stoppers

lrishi committed:
SSPROD-23034: Expand /var/run volume mount to include entire /var

Agent requires access to container fs for internal state validation
and feature support. On some platforms, these directories might
reside outside of /var/run in /var/<vendor-specific> directories.

lrishi committed:
SSPROD-23034: Expand /var/run volume mount to include entire /var

Agent requires access to container fs for internal state validation
and feature support. On some platforms, these directories might
reside outside of /var/run in /var/<vendor-specific> directories.

mavimo commented on charts/agent/templates/deployment.yaml:
```diff
@@ -184,9 +184,9 @@ spec:
           name: run-vol
         - hostPath:
-            path: /var/run
+            path: /var
```
The deployment is intended only for collecting k8s infos, do we need to get access to the whole `/var` folder also in this case?

lrishi replied:
If that is the case, then no, but since its agent deployment, my understanding is that both are 1/1 mapped, except for Deployment/Daemonset, but I might be wrong. There is some value to keeping both in sync, but I'm happy to remove this part, if you suggest so.

mavimo replied:
@lrishi IMHO exposing information that are not relevant into container may be not needed, at the same time it may increase maintenance complexity; let ask for a 3rd opinion from @aroberts87

aroberts87 replied:
Hey @lrishi - I agree with @mavimo and I believe you can remove the change to map all of `/var` into the Deployment agents. Those agents are running in `nodriver` mode and do not send syscall metrics.

lrishi committed:
Remove changes to deployment

mavimo committed:
feat(): add unit test for volume mount

mavimo committed:
chore(): temporary disable test on volumes

mavimo committed:
chore(agent): Bump chart version

mavimo committed:
chore(agent): mount only /var/lib and /var/run in container

Mounting the /var folder, when /var/run is a symlink to /run the link
is resolved inside the container, this not allow to get access to the
host information.

mavimo committed:
chore(agent, sysdig): update chart version.

mavimo committed:
chore(agent): update volumes shared in the deployments

mavimo commented:
The scope of this PR is changed, during tests some issue emerged.

Mounting the whole `/var` folder may create issues related to symlinks (e.g.: `/var/run -> /run` we ending up by exposing in the `/host/var/run` the `/run` folder in the container instead of the `/run` folder of the host).
Since @lrishi report that only the `/var/lib` folder is required to be exposed in the `/host/var/lib` folder we limit the mount of this specific folder in order to limit issues.
